src/wallet.c

```diff
@@ -198,15 +198,26 @@ int wallet_master_from_mnemonic(char *mnemo, int m_len, const char *salt, int s_
 int wallet_generate_key(HDNode *node, const char *keypath, int keypath_len,
                         const uint8_t *privkeymaster, const uint8_t *chaincode)
 {
-    unsigned long idx = 0;
+    static char delim[] = "/";
+    static char prime[] = "phH\'";
+    static char digits[] = "0123456789";
+    uint32_t idx = 0;
     char *pch, *kp = malloc(keypath_len + 1);
 
     if (!kp) {
         return STATUS_ERROR_MEM;
     }
 
+    if (keypath_len < 2) {
+        goto err;
+    }
+
+    memset(kp, 0, keypath_len + 1);
```

Comment:
Added a length check here and also zeroed out the allocated memory (and rebased).